### PR TITLE
feat(macros): init

### PR DESCRIPTION
**Pull request by W95Psp**

### CODE REVIEW

File: src/contributing/libraries_macros.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+OCaml, thus there is no manual parsing involved.
+
+## User experience
+Asking the user to type `#[_hax::json(some_long_json)]` is not very
+friendly. Thus, the crate `hax-lib-macros` defines a bunch of [proc
+macros](https://doc.rust-lang.org/beta/reference/procedural-macros.html)
+that defines nice and simple-to-use macros. Those macro take care of
+cooking some `hax_lib_macros_types::AttrPayload` payload(s), then
+serialize those payloads to JSON and produce one or more
+`#[_hax::json(serialized_pyaload)]` attributes.
```
Comment (jschneider-bensch) [inline src/contributing/libraries_macros.md:24]:
```suggestion
`#[_hax::json(serialized_payload)]` attributes.
```

File: src/contributing/libraries_macros.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+The hax engine understands only one attribute: `#[_hax::json(PAYLOAD)]`,
+where `PAYLOAD` is a JSON serialization of the Rust enum
+`hax_lib_macros_types::AttrPayload`.
+
+Note `#[_hax::json(PAYLOAD)]` is a [tool
+attribute](https://github.com/rust-lang/rust/issues/66079): an
+attribute that is never expanded.
+
+In the engine, the OCaml module `Attr_payloads` offers an API to query
+attributes easily. The types in crate `hax_lib_macros_types` and
```
Comment (jschneider-bensch) [inline src/contributing/libraries_macros.md:13]: Could you give an example how to use the API in `Attr_payloads`?

**Review by jschneider-bensch — COMMENTED** [review]
Thanks for writing this down!

File: src/contributing/libraries_macros.md (hunk 2 above)
Comment (W95Psp) [reply src/contributing/libraries_macros.md:13]: Sure, I'm gonna write something now

**Review by franziskuskiefer — APPROVED** [review]
If @jschneider-bensch is fine with this, let's get this in.

**Comment by W95Psp** [comment]
Ok. That's better I think, I want to document that directly on the `Attr_payloads` module (this partially motivated https://github.com/hacspec/hax/pull/361)

**Comment by jschneider-bensch** [comment]
:+1: That sounds good! Fine to merge, from my side.

**Pull request merged**